lib/nmap.ex

```diff
@@ -42,9 +42,17 @@ defmodule Hades.NMAP do
     Process.flag(:trap_exit, true)
 
     path = Helpers.hades_path()
+  
+    target_vector = String.splitter("#{target}", ["/"]) |> Enum.take(2)
+    command = if (length(target_vector) == 2) do 
+      subnet = Enum.at(target_vector, 1)
+      "nmap #{option} -oX #{path} #{target} / #{subnet}"
+    else
+      "nmap #{option} -oX #{path} #{target}"
+    end
 
     port =
-      Port.open({:spawn, "nmap #{option} -oX #{path} #{target}"}, [
+      Port.open({:spawn, command}, [
```

Comment:
I think the code would look cleaner if you continue to use the old "structure" `nmap #{option} -oX #{path}...`.  
Just use the `target` variable to compose the given `#{target} / #{subnet}` or `#{target}`.

Another option that just came in my mind would be to build the target string in the `hades.ex` module so you don't have to handle this here. I think this would reduce the overall loc a bit.

Reply:
I agree on making the code cleaner. Will work on this.